docs/discovery.md

```diff
@@ -98,8 +98,14 @@ You can list the nodes in your cluster.
     <node_ip2:2375>
     <node_ip3:2375>
 
+### Using a Distributed Key/Value Store
 
-### Using etcd
+Distributed Key/Value Store discovery is the recommended way to handle
+the node discovery in Swarm. This is supported through [libkv](https://github.com/docker/libkv),
+you can use any of the distributed k/v backend supported by the library
+(which as of now is limited to either Consul/etcd or zookeeper).
```

Comment:
not sure I'd mention what it's limited to. Perhaps, "For an overview of the supported backends, refer to the [libkv documentation](https://github.com/docker/libkv)